contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,146 @@
+import {IBlackScholes} from "../interfaces/IBlackScholes.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {Math} from "../libraries/Math.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import "hardhat/console.sol";
+
+contract OptionsPremiumPricer is DSMath {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IVolatilityOracle public immutable volatilityOracle;
+    IPriceOracle public immutable priceOracle;
+    ICompToken public immutable cToken;
+    // IKEEP3rVolatility: 0xCCdfCB72753CfD55C5afF5d98eA5f9C43be9659d
+    IBlackScholes public immutable blackScholes;
+
+    constructor(
+        address _volatilityOracle,
+        address _priceOracle,
+        address _cToken,
+        address _blackScholes
+    ) {
+        require(_volatilityOracle != address(0), "!_volatilityOracle");
+        require(_priceOracle != address(0), "!_priceOracle");
+        require(_cToken != address(0), "!_cToken");
+        require(_blackScholes != address(0), "!_blackScholes");
+        volatilityOracle = IVolatilityOracle(_volatilityOracle);
+        priceOracle = IPriceOracle(_priceOracle);
+        cToken = ICompToken(_cToken);
+        blackScholes = IBlackScholes(_blackScholes);
+    }
+
+    /**
+     * @notice Calculates the premium of the provided option using Black-Scholes
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @param isPut is whether the option is a put option
+     */
+    function getPremium(
+        uint256 st,
+        uint256 expiryTimestamp,
+        bool isPut
+    ) external view returns (uint256 premium) {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        uint256 sp = priceOracle.latestAnswer().div(10**priceOracle.decimals());
+        uint256 v = volatilityOracle.vol();
+        uint256 t = expiryTimestamp.sub(block.timestamp).div(1 days);
+
+        console.log("strike price is %s", st);
+        console.log("underlying price is %s", sp);
+        console.log("volatility is %s", v);
+        console.log("time to expiry in days is %s", t);
+
+        (uint256 call, uint256 put) = blackScholes.quoteAll(t, v, sp, st);
+        console.log("call option price is %s", call);
+        console.log("put option price is %s", put);
+
+        premium = !isPut ? call : put;
+    }
+
+    /**
+     * @notice Calculates the risk free interest rate on the premium underlying asset
+     * using Compounds's risk free interest rate
+     * Formula reference: https://compound.finance/docs (Calculating the APY Using Rate Per Block)
+     */
+    function getAssetRiskFreeRate() private view returns (uint256 apy) {
```

Comment:
getAssetRiskFreeRate entails some tricky math for getting decimals right - its not correct yet. will either work out it out if we need to use or delete it

Reply:
We should just delete it imo